src/robotFactory.js

```diff
@@ -1,10 +1,67 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = coords === undefined || coords.x === undefined ? {
+      x: 0,
+      y: 0,
+    } : coords;
```

Comment:
```suggestion
    this.coords = {
      x: coords.x || 0,
      y: coords.y || 0,
    };
```

Давай спростимо цей запис